### ISSUE TITLE
Factor out all deps except Docker

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/onpremise#999]
We use Docker to run Sentry in a platform-neutral way, but the installation script depends on non-Docker tools such as `bash`, `awk`, `sed`, etc., which leads to portability bugs such as https://github.com/getsentry/onpremise/issues/941. There we have a [suggestion](https://github.com/getsentry/onpremise/issues/941#issuecomment-862570423) to remove all non-Docker dependencies from the installation flow. Seems like a worthy goal to me.

# Todo

- [ ] audit for non-Docker deps
- [ ] ???

**Comment by Daniel15** [comment]
One potential approach that I don't think would be too much work is to have a small install kickstart script that just checks for Docker (just a basic check that `docker --version` works or something like that) then runs a Docker container that runs the actual installation script. You could just mount the current directory as a bind mount in the installation container.

Once that's done, you could probably allow installation on Docker on Windows by having a batch or PowerShell kickstart script that does the same thing (checks for Docker then runs the installation container).

**Comment by ethanhs** [comment]
Things we do on the host that should(?) be moved into a container:

- docker compose run (?)
- docker compose build (?)
- docker compose pull (?)
- docker compose stop (?) (used for cleanup)
- docker run/docker pull (?) for various one off containers or actions (e.g. curl)
- git version check
- sort/sed/etc for version checking
- volume creation?
- docker info for platform checks
- jq related to relay credentials
- head/tr for setting the secret for sentry config
- diff for checking geoip's database
- grep for extracting data
- cp to move files
- probably a couple of other things I've missed


Various important things we do to consider the installer has to do:
- set up kafka, tsdb, geoip, zookeeper, clickhouse
- build images
- set up snuba
- make volumes
- update images
- check the install worked correctly

I'm going to keep thinking about the best way to factor out all the dependencies to just require a container runtime e.g. docker or podman.

**Comment by Charismara** [comment]
Hey there,is there any progress on this?

**Comment by aldy505** [comment]
> Hey there,is there any progress on this?

@Charismara I don't think this will ever be worked on, since it's hard to maintain all dependencies (Kafka, ClickHouse, Postgres, Redis, Memcached) being external. This means we'll have more moving parts.
